cores/esp8266/core_esp8266_si2c.c

```diff
@@ -17,14 +17,90 @@
   You should have received a copy of the GNU Lesser General Public
   License along with this library; if not, write to the Free Software
   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+  
+  Modified January 2017 by Bjorn Hammarberg (bjoham@esp8266.com) - i2c slave support
 */
 #include "twi.h"
 #include "pins_arduino.h"
 #include "wiring_private.h"
 
+//#include <compat/twi.h>
+#include "twi_util.h"
+
+#include "ets_sys.h"
+
 unsigned char twi_dcount = 18;
 static unsigned char twi_sda, twi_scl;
 static uint32_t twi_clockStretchLimit;
+static unsigned char twi_addr = 0;
+
+// modes (private)
+#define TWIPM_UNKNOWN 	0
+#define TWIPM_IDLE 		1
+#define TWIPM_ADDRESSED	2
+#define TWIPM_WAIT		3
+
+// states (private)
+#define TWIP_UNKNOWN 	0
+#define TWIP_IDLE 		1
+#define TWIP_START 		2
+#define TWIP_SEND_ACK	3
+#define TWIP_WAIT_ACK	4
+#define TWIP_WAIT_STOP	5
+#define TWIP_SLA_W		6
+#define TWIP_SLA_R		7
+#define TWIP_REP_START	8
+#define TWIP_READ		9
+#define TWIP_STOP 		10
+#define TWIP_REC_ACK	11
+#define TWIP_READ_ACK	12
+#define TWIP_RWAIT_ACK	13
+#define TWIP_WRITE		14
+#define TWIP_BUS_ERR 	15
```

Comment:
It would be great to describe what each of these states means, and what are the possible transitions. Looking at the code, it seems to me that some of the states here are not actually being used. This does not help understand how the proposed code works.